### PR TITLE
Splitting Host.Client

### PR DESCRIPTION
**Pull request by karthiknadig**
This PR splits Host.Client with minimal code change. Moving more pieces from windows specific to x-plat will require significant code change and should be done in the next pass.

### CODE REVIEW

File: src/R/Components/Test/Microsoft.R.Components.Test.csproj [hunk 1]
```diff
@@ -190,6 +182,14 @@
       <Project>{1f89902f-853e-44b5-93a7-e237984c4fb7}</Project>
       <Name>Microsoft.Windows.Core.Test</Name>
     </ProjectReference>
+    <ProjectReference Include="..\..\..\Windows\Host\Client\Impl\Microsoft.R.Host.Client.Windows.csproj">
```
Comment (MikhailArkhipov) [inline src/R/Components/Test/Microsoft.R.Components.Test.csproj]: What does Components need from the Windows part? I am splitting Components next.

**Comment by MikhailArkhipov** [comment]
Looks like just files moved, no logic changes?

**Comment by AlexanderSher** [comment]
Why `RHost` had to be moved to Windows?

**Comment by karthiknadig** [comment]
This PR has no logic changes. Hence, there is still code such as RHost and Broker services still on windows side. I am moving stuff out to x-plat binary with code changes as we speak.

**Comment by MikhailArkhipov** [comment]
OK. 👍

**Pull request merged**